pkg/utils/grpc_test.go

```diff
@@ -0,0 +1,75 @@
+package utils
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/spf13/pflag"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/openservicemesh/osm/pkg/certificate"
+	"github.com/openservicemesh/osm/pkg/certificate/providers/tresor"
+	"github.com/openservicemesh/osm/pkg/constants"
+)
+
+func TestNewGrpc(t *testing.T) {
+	assert := assert.New(t)
+	cache := make(map[certificate.CommonName]certificate.Certificater)
+	certManager := tresor.NewFakeCertManager(&cache, 1*time.Hour)
+	adsCert, err := certManager.GetRootCertificate()
+	assert.Nil(err)
+
+	certPem := adsCert.GetCertificateChain()
+	keyPem := adsCert.GetPrivateKey()
+	rootPem := adsCert.GetIssuingCA()
+	emptyByteArray := []byte{}
+
+	type newGrpcTest struct {
+		serverType string
+		port       int
+		certPem    []byte
+	}
+
+	newGrpcTests := []newGrpcTest{
+		{"abc", 123, emptyByteArray},
+		{"ADS", 8081, emptyByteArray},
+		{"ADS", 8080, certPem},
+	}
+
+	for _, gt := range newGrpcTests {
+		resServer, resListener, err := NewGrpc(gt.serverType, gt.port, gt.certPem, keyPem, rootPem)
+		if err != nil {
```

Comment:
Please add the expected err value to newGrpcTest so that we can check that the correct tests are erroring